### ISSUE TITLE
Feature request: provide example for constant header parameters

### ISSUE THREAD

**Issue by kubukoz** [softwaremill/tapir#652]
Hi! First of all, thanks for the amazing library, I love it.

Here's a simple endpoint that takes a constant header value:

```scala
object HeaderRepro extends App {
  import sttp.tapir.docs.openapi._
  import sttp.tapir.openapi.circe.yaml._

  import sttp.tapir._

  val endpoint = infallibleEndpoint
    .in("examples")
    .in("hello")
    .in(header("Content-Type", "application/json"))

  println(
    List(endpoint)
      .toOpenAPI("Title", "version")
      .toYaml
  )
}
```

The generated yaml looks like this:

```yaml
openapi: 3.0.1
info:
  title: Title
  version: version
paths:
  /examples/hello:
    get:
      operationId: getExamplesHello
      parameters:
      - name: Content-Type
        in: header
        required: true
        schema:
          type: string
      responses:
        '200':
          description: ''
```

It's impossible to add an example to the header input, because it outputs `Unit` (as it's constant). I think for constant headers (or constant inputs, even) we should be able to see the required value as an example in the documentation.

**Comment by adamw** [comment]
Fixed in 0.16.4

**Comment by kubukoz** [comment]
Thanks, that was fast!

I'm afraid it won't really help me a lot though, as apparently swagger-ui doesn't send Content-Type in GET requests...
